event/store.go

```diff
@@ -0,0 +1,60 @@
+package event
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+)
+
+const defaultEventCountLimit = 5
+
+// Store stores events
+type Store struct {
+	*sync.RWMutex
+
+	events map[string][]*Event // taskname => events
+	limit  int
+}
+
+// NewStore returns a new store
+func NewStore() *Store {
+	return &Store{
+		RWMutex: &sync.RWMutex{},
```

Comment:
For precautionary measures, the mux would be more controlled as a private attribute so callers of the NewStore could not maliciously hold onto a lock outside of the context of the store methods.

Reply:
good point! let me do this!